[kokrui] iP

kokrui opened the pull request:
# a heading

- a bullet list
- a bullet list

1. a numbered list 
2. a numbered list 

```python
print("a fenced code block with syntax highlighting")
```

- [x] a task list
- [ ] wow task

🅰️n emoji

> a block quote

[a hyperlink](https://github.com/nus-cs2103-AY2324S1/ip/pull/10)

some `inline` code

**bold** *italics* ~~strikethrough~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

kokrui committed:
feat: Add greet and exit messages when run

kokrui committed:
feat: Add helper function for lines and indentation

kokrui committed:
refactor: Modularize greet and exit methods

kokrui committed:
feat: Echo user input until 'bye' received

kokrui committed:
feat: Add list feature

kokrui committed:
feat: Implement Task class

kokrui committed:
feat: Support marking and unmarking

kokrui committed:
refactor: Make Task attributes protected

kokrui committed:
feat: Implement Deadline subclass

kokrui committed:
fix: Change Deadline field name and access

kokrui committed:
feat: Implement Event subclass

kokrui committed:
feat: Implement Todo subclass

kokrui committed:
feat: Print list size when task added

kokrui committed:
feat: Add command support for adding different task types

kokrui committed:
test: Add initial text ui test cases

kokrui committed:
feat: Create base DukeException class

kokrui committed:
feat: Add exception handling to input parser

kokrui committed:
test: Update tests to include expected error messages

kokrui committed:
refactor: Delegate Task creation to subclass factory methods

kokrui committed:
docs: Add docstrings for subclass factory methods

kokrui committed:
refactor: Cleanup prints and make attributes final

kokrui committed:
feat: Add delete command to delete task

kokrui committed:
test: Include delete command in tests

kokrui committed:
feat: Add FileUtils class with save/read methods

kokrui committed:
feat: Add (de)serialization code for each Task subclass

kokrui committed:
feat: Load tasks from file on startup

kokrui committed:
feat: Save tasks to file on list update

kokrui committed:
chore: Add data directories to gitignore

kokrui committed:
fix: Add space between event name and time info

kokrui committed:
fix: Prevent pipes in input which corrupts save files

kokrui committed:
fix: Handle save file corrupt errors gracefully

kokrui committed:
Merge branch 'branch-Level-7'

kokrui committed:
test: Update tests with file load message + missing space

kokrui committed:
feat: implement date parsing for Deadline

kokrui committed:
feat: print deadline by LocalDate in different format

kokrui committed:
feat: Add date parsing to Event

kokrui committed:
fix: Add missing space to Event toString

kokrui committed:
Merge branch 'branch-Level-8'

ruishanteo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+        this.isDone = false;
+    }
+
+    public static Task fromFileFormat(String line) throws DukeException {
```
Should this method have a javadoc?

kokrui replied:
Thanks! I agree that it should have additional documentation due to its complexity and I've added the JavaDoc

ruishanteo commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,82 @@
+     * @return A new Event instance.
+     * @throws DukeException If the input format is invalid.
+     */
+    public static Event createFromCommandString(String input) throws DukeException {
```
Will you be adding time on top of dates, as per Level 8?

ruishanteo commented on src/main/java/FileUtils.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class FileUtils {
```
I think this class is missing header comments, as well as method documentations.

kokrui replied:
Thanks! I agree that it should have a header comment since it's a public class so I've added it

ruishanteo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+     */
+    @Override
+    public String toString() {
+        return "[" + (isDone ? "X" : " ") + "] " + this.name;
```
Maybe you can consider using String.format for this.

ruishanteo reviewed (commented):
Overall I find your code easy to read and executed very well!

ruishanteo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+public abstract class Task {
```
Should all files be in packages?

kokrui replied:
Thanks! I agree that all classes should belong to packages so I've since made that change

ruishanteo commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException extends Exception {
```
Header comments would be good to have for public classes.

kokrui replied:
Thanks! I agree that it should have a header comment since it's a public class so I've added it

ruishanteo reviewed (commented):
All the files look good, except for some missing documentation!

kokrui committed:
refactor: Abstract away user interactions into Ui class

kokrui committed:
refactor: Update Duke to support Ui class

kokrui committed:
refactor: Abstract file i/o tasks to Storage class

kokrui committed:
refactor: Abstract task list functionality to TaskList

kokrui committed:
refactor: Abstract input parsing through Parser class

kokrui committed:
test: Update tests with new output format

kokrui committed:
Merge branch 'A-MoreOOP'

kokrui committed:
refactor: Add files to koko package

kokrui committed:
test: Update test scripts to reflect new package

kokrui committed:
Merge branch 'A-Packages'

kokrui committed:
Merge branch 'add-gradle-support' into A-Gradle

kokrui committed:
fix: Add correct package name to gradle file

kokrui committed:
Merge branch 'A-Gradle'

kokrui committed:
test: Add JUnit tests for TaskList and Parser

kokrui committed:
Merge branch 'A-JUnit'

kokrui committed:
feat: Verify that JAR file creation works

kokrui committed:
Merge branch 'A-Jar'

kokrui committed:
docs: Add JavaDocs to all methods and classes

kokrui committed:
Merge branch 'A-JavaDoc'

kokrui committed:
style: Unindent cases in switch statements

kokrui committed:
style: Break long lines

kokrui committed:
Merge branch 'A-CodingStandard'

kokrui committed:
feat: Add support for find command

kokrui committed:
test: Update text ui tests for find

kokrui committed:
Merge branch 'Level-9'

kokrui committed:
fix: Add missing space in error message

kokrui committed:
feat: Add JavaFX as dependency in gradle config

kokrui committed:
feat: Add images and fxml views in resources dir

kokrui committed:
feat: Add JavaFX skeleton code to show chat UI

kokrui committed:
feat: Connect UI to Duke tasklist logic

kokrui committed:
Merge branch 'Level-10'

kokrui committed:
feat: Add assertions for startup message

kokrui committed:
Merge pull request #2 from kokrui/branch-A-Assertions

feat: Add assertions for startup message

kokrui committed:
refactor: Abstract away individual Command functionality

kokrui committed:
refactor: Remove useless throws statement

kokrui committed:
Merge branch 'master' into branch-A-CodeQuality

kokrui committed:
Merge pull request #3 from kokrui/branch-A-CodeQuality

Improve Code Quality

kokrui committed:
feat: Add flexibility to task search

Previously, the task search functionality required the given
input user keyword string to be an exact match to any substring
of a task name.

This commit allows for better flexibility by:

* firstly, making the search function case-insensitive
* secondly, trimming the keyword input to prevent any leading
  or lagging whitespace to impact sensitivity of the search.

Which are both reasonable mistakes that users may make
when attempting to search for a specific set of tasks.

kokrui committed:
Merge branch 'master' into branch-C-BetterSearch

kokrui committed:
Merge pull request #4 from kokrui/branch-C-BetterSearch

Improve search flexibility

kokrui committed:
fix: Remove bug that broke BYE command

kokrui committed:
docs: Add product screenshot

kokrui committed:
docs: Update product screenshot

kokrui committed:
docs: Add user guide

kokrui committed:
fix: Update build gradle to build cross-platform JARs

kokrui committed:
fix: Remove error from fresh data file message

kokrui committed:
refactor: Add spaces in javadoc sections, remove unnecessary spaces

kokrui committed:
docs: Add Javadocs for remaining public classes/methods

kokrui committed:
fix: Add missing newline between quotes
